metis-enrichment/metis-enrichment-common/src/main/java/eu/europeana/enrichment/api/internal/AbstractSearchTerm.java

```diff
@@ -0,0 +1,30 @@
+package eu.europeana.enrichment.api.internal;
+
+import eu.europeana.enrichment.utils.EntityType;
+import eu.europeana.metis.schema.jibx.LanguageCodes;
+import java.util.List;
+
+public abstract class AbstractSearchTerm implements SearchTerm {
+
+  private String textValue;
+  private LanguageCodes language;
+
+  public AbstractSearchTerm(String textValue, LanguageCodes language){
+    this.textValue = textValue;
+    this.language = language;
+  }
+
+  public abstract List<EntityType> getCandidateTypes();
+
+  public abstract boolean equals(SearchTerm searchTerm);
+
+  public abstract int hashCode();
```

Comment:
I understand what you mean, and I think that it does make to use `textValue` and `language` since they are the "main" values, but that would lead to many objects with the same hashCode value even if they have different `fieldTypes`.
Should we follow the same reasoning for `equals()` then?
Nevertheless I can implement the `hashCode()` in the abstract classes and later if we think the subclasses should be responsible for this, then we can use `super()`